File: .sdk-repos/version-8.9/orchestration-cluster-api-csharp/examples/ProcessDefinition.cs
// Compilable usage examples for process definition operations.
// These examples are type-checked during build to guard against API regressions.
using Camunda.Orchestration.Sdk;

public static class ProcessDefinitionExamples
{
    #region GetProcessDefinition
    // <GetProcessDefinition>
    public static async Task GetProcessDefinitionExample(ProcessDefinitionKey processDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionAsync(
            processDefinitionKey);

        Console.WriteLine($"Process definition: {result.Name}");
    }
    // </GetProcessDefinition>
    #endregion GetProcessDefinition

    #region GetProcessDefinitionXml

    // <GetProcessDefinitionXml>
    public static async Task GetProcessDefinitionXmlExample(ProcessDefinitionKey processDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionXmlAsync(
            processDefinitionKey);

        Console.WriteLine($"XML: {result}");
    }
    // </GetProcessDefinitionXml>
    #endregion GetProcessDefinitionXml

    #region SearchProcessDefinitions

    // <SearchProcessDefinitions>
    public static async Task SearchProcessDefinitionsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.SearchProcessDefinitionsAsync(
            new ProcessDefinitionSearchQuery());

        foreach (var pd in result.Items)
        {
            Console.WriteLine($"Process definition: {pd.Name}");
        }
    }
    // </SearchProcessDefinitions>
    #endregion SearchProcessDefinitions

    #region GetProcessDefinitionStatistics

    // <GetProcessDefinitionStatistics>
    public static async Task GetProcessDefinitionStatisticsExample(ProcessDefinitionKey processDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionStatisticsAsync(
            processDefinitionKey,
            new ProcessDefinitionElementStatisticsQuery());

        foreach (var stat in result.Items)
        {
            Console.WriteLine($"Element: {stat.ElementId}");
        }
    }
    // </GetProcessDefinitionStatistics>
    #endregion GetProcessDefinitionStatistics

    #region GetProcessDefinitionInstanceStatistics

    // <GetProcessDefinitionInstanceStatistics>
    public static async Task GetProcessDefinitionInstanceStatisticsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionInstanceStatisticsAsync(
            new ProcessDefinitionInstanceStatisticsQuery());

        foreach (var stat in result.Items)
        {
            Console.WriteLine($"Definition: {stat.ProcessDefinitionId}");
        }
    }
    // </GetProcessDefinitionInstanceStatistics>
    #endregion GetProcessDefinitionInstanceStatistics

    #region GetProcessDefinitionInstanceVersionStatistics

    // <GetProcessDefinitionInstanceVersionStatistics>
    public static async Task GetProcessDefinitionInstanceVersionStatisticsExample(ProcessDefinitionId processDefinitionId)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionInstanceVersionStatisticsAsync(
            new ProcessDefinitionInstanceVersionStatisticsQuery
            {
                Filter = new ProcessDefinitionInstanceVersionStatisticsFilter
                {
                    ProcessDefinitionId = processDefinitionId,
                },
            });

        foreach (var stat in result.Items)
        {
            Console.WriteLine($"Version: {stat.ProcessDefinitionVersion}");
        }
    }
    // </GetProcessDefinitionInstanceVersionStatistics>
    #endregion GetProcessDefinitionInstanceVersionStatistics

    #region GetProcessDefinitionMessageSubscriptionStatistics

    // <GetProcessDefinitionMessageSubscriptionStatistics>
    public static async Task GetProcessDefinitionMessageSubscriptionStatisticsExample()
    {
        using var client = CamundaClient.Create();

        var result = await client.GetProcessDefinitionMessageSubscriptionStatisticsAsync(
            new ProcessDefinitionMessageSubscriptionStatisticsQuery());

        foreach (var stat in result.Items)
        {
            Console.WriteLine($"Message subscriptions: {stat.ActiveSubscriptions}");
        }
    }
    // </GetProcessDefinitionMessageSubscriptionStatistics>
    #endregion GetProcessDefinitionMessageSubscriptionStatistics

    #region GetStartProcessForm

    // <GetStartProcessForm>
    public static async Task GetStartProcessFormExample(ProcessDefinitionKey processDefinitionKey)
    {
        using var client = CamundaClient.Create();

        var result = await client.GetStartProcessFormAsync(
            processDefinitionKey);

        Console.WriteLine($"Form: {result.FormKey}");
    }
    // </GetStartProcessForm>
    #endregion GetStartProcessForm
}

File: .sdk-repos/orchestration-cluster-api-php/examples/process_definition.php
<?php

/**
 * Compilable usage examples for process-definition operations.
 */

declare(strict_types=1);

namespace Camunda\Orchestration\Examples;

use Camunda\Orchestration\CamundaClient;
use Camunda\Orchestration\Semantic\ProcessDefinitionKey;

// region SearchProcessDefinitions
/**
 * Search process definitions.
 */
function search_process_definitions(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\ProcessDefinitionSearchQuery $processDefinitionSearchQuery = null): void
{
    $client->searchProcessDefinitions($processDefinitionSearchQuery);
}
// endregion SearchProcessDefinitions

// region GetProcessDefinitionMessageSubscriptionStatistics
/**
 * Get message subscription statistics.
 */
function get_process_definition_message_subscription_statistics(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\ProcessDefinitionMessageSubscriptionStatisticsQuery $processDefinitionMessageSubscriptionStatisticsQuery = null): void
{
    $client->getProcessDefinitionMessageSubscriptionStatistics($processDefinitionMessageSubscriptionStatisticsQuery);
}
// endregion GetProcessDefinitionMessageSubscriptionStatistics

// region GetProcessDefinitionInstanceStatistics
/**
 * Get process instance statistics.
 */
function get_process_definition_instance_statistics(CamundaClient $client, ?\Camunda\Orchestration\Api\Model\ProcessDefinitionInstanceStatisticsQuery $processDefinitionInstanceStatisticsQuery = null): void
{
    $client->getProcessDefinitionInstanceStatistics($processDefinitionInstanceStatisticsQuery);
}
// endregion GetProcessDefinitionInstanceStatistics

// region GetProcessDefinition
/**
 * Get process definition.
 */
function get_process_definition(CamundaClient $client, ProcessDefinitionKey $processDefinitionKey): void
{
    $client->getProcessDefinition((string) $processDefinitionKey);
}
// endregion GetProcessDefinition

// region GetStartProcessForm
/**
 * Get process start form.
 */
function get_start_process_form(CamundaClient $client, ProcessDefinitionKey $processDefinitionKey): void
{
    $client->getStartProcessForm((string) $processDefinitionKey);
}
// endregion GetStartProcessForm

// region GetProcessDefinitionStatistics
/**
 * Get process definition statistics.
 */
function get_process_definition_statistics(CamundaClient $client, ProcessDefinitionKey $processDefinitionKey, ?\Camunda\Orchestration\Api\Model\ProcessDefinitionElementStatisticsQuery $processDefinitionElementStatisticsQuery = null): void
{
    $client->getProcessDefinitionStatistics((string) $processDefinitionKey, $processDefinitionElementStatisticsQuery);
}
// endregion GetProcessDefinitionStatistics

// region SearchProcessDefinitionVariableNames
/**
 * Search process definition variable names.
 */
function search_process_definition_variable_names(CamundaClient $client, ProcessDefinitionKey $processDefinitionKey, ?\Camunda\Orchestration\Api\Model\ProcessDefinitionVariableNameSearchQuery $processDefinitionVariableNameSearchQuery = null): void
{
    $client->searchProcessDefinitionVariableNames((string) $processDefinitionKey, $processDefinitionVariableNameSearchQuery);
}
// endregion SearchProcessDefinitionVariableNames

// region GetProcessDefinitionXML
/**
 * Get process definition XML.
 */
function get_process_definition_xml(CamundaClient $client, ProcessDefinitionKey $processDefinitionKey): void
{
    $client->getProcessDefinitionXML((string) $processDefinitionKey);
}
// endregion GetProcessDefinitionXML

// region GetProcessDefinitionInstanceVersionStatistics
/**
 * Get process instance statistics by version.
 */
function get_process_definition_instance_version_statistics(CamundaClient $client, \Camunda\Orchestration\Api\Model\ProcessDefinitionInstanceVersionStatisticsQuery $processDefinitionInstanceVersionStatisticsQuery): void
{
    $client->getProcessDefinitionInstanceVersionStatistics($processDefinitionInstanceVersionStatisticsQuery);
}
// endregion GetProcessDefinitionInstanceVersionStatistics
